### PR TITLE
Convert octahedral compressed normals/tangents on 3.x meshes

### PR DESCRIPTION
**Pull request by nikitalita**
When Octahedral compression was added to the 3.x branch (https://github.com/godotengine/godot/pull/46800), the functionality to convert 3.x meshes upon import was not updated. As a result, it is broken on meshes that have Octahedral compression enabled (this is enabled by default in 3.4).

This PR adds converting octahedral compressed normals and tangents in old 3.x format meshes back into their uncompressed form upon import.

### CODE REVIEW

**Review by Calinou — COMMENTED** [review]
You added `modules/gdsdecomp` by accident in this PR – it should be removed.

**Comment by nikitalita** [comment]
fixed.

**Comment by clayjohn** [comment]
Cc @The-O-King

**Review by The-O-King — APPROVED** [review]
This all looks great to me as well! Hopefully we'll have octahedral normals/tangent supported in 4.x soon as well too!

File: scene/resources/mesh.cpp [hunk 1]
```diff
@@ -626,6 +626,27 @@ static Mesh::PrimitiveType _old_primitives[7] = {
 };
 #endif // DISABLE_DEPRECATED
 
+// Convert Octohedron-mapped normalized vector back to Cartesian
+// Assumes normalized format (elements of v within range [-1, 1])
```
Comment (akien-mga) [inline scene/resources/mesh.cpp:630]:
```suggestion
// Convert Octahedron-mapped normalized vector back to Cartesian.
// Assumes normalized format (elements of v within range [-1, 1]).
```

File: scene/resources/mesh.cpp [hunk 2]
```diff
@@ -626,6 +626,27 @@ static Mesh::PrimitiveType _old_primitives[7] = {
...
+	Vector3 res(v.x, v.y, 1 - (Math::absf(v.x) + Math::absf(v.y)));
+	float t = MAX(-res.z, 0.0f);
+	res.x += t * -SIGN(res.x);
+	res.y += t * -SIGN(res.y);
+	return res.normalized();
+}
+
+// Convert Octohedron-mapped normalized tangent vector back to Cartesian
+// out_sign provides the direction for the original cartesian tangent
+// Assumes normalized format (elements of v within range [-1, 1])
```
Comment (akien-mga) [inline scene/resources/mesh.cpp:641]:
```suggestion
// Convert Octahedron-mapped normalized tangent vector back to Cartesian.
// out_sign provides the direction for the original Cartesian tangent.
// Assumes normalized format (elements of v within range [-1, 1]).
```

File: scene/resources/mesh.cpp [hunk 3]
```diff
@@ -692,66 +713,138 @@ void _fix_array_compatibility(const Vector<uint8_t> &p_src, uint32_t p_old_forma
 				}
 			} break;
 			case OLD_ARRAY_NORMAL: {
-				if (p_old_format & OLD_ARRAY_COMPRESS_NORMAL) {
-					const float multiplier = 1.f / 127.f * 1023.0f;
+				if (p_old_format & OLD_ARRAY_FLAG_USE_OCTAHEDRAL_COMPRESSION) {
+					if ((p_old_format & OLD_ARRAY_COMPRESS_NORMAL) && (p_old_format & OLD_ARRAY_FORMAT_TANGENT) && (p_old_format & OLD_ARRAY_COMPRESS_TANGENT)) {
+						for (uint32_t i = 0; i < p_elements; i++) {
+							//const float multiplier = 1.f / 127.f * 1023.0f;
```
Comment (akien-mga) [inline scene/resources/mesh.cpp:719]: Dead code to remove (added four times always commented out).

**Comment by nikitalita** [comment]
Fixed

**Comment by akien-mga** [comment]
There's a third-party submodule that should be removed.

**Comment by nikitalita** [comment]
fixed

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**